compaction/compaction.cc

```diff
@@ -619,6 +664,7 @@ class compaction {
     compaction& operator=(compaction&& other) = delete;
 
     virtual ~compaction() {
+        _monitor_generator.reset_generator();
     }
```

Comment:
we need to reset because of retry right? anyway it's the right thing to do.

Reply:
Yes, and not to keep a generator alive while it won't be used anymore